tests/lambda_functions/va_profile/test_va_profile_integration.py

```diff
@@ -0,0 +1,247 @@
+"""
+notify_db and notify_db_session are fixtures in tests/conftest.py.
+
+https://docs.sqlalchemy.org/en/13/core/connections.html
+
+Truncating the va_profile_local_cache table at the beginning and end of some tests is necessary because the database
+side effects of executing the VA Profile lambda function are not rolled back at the conclusion of a test.
+"""
+
+import os
+from lambda_functions.va_profile.va_profile_opt_in_out_lambda import va_profile_opt_in_out_lambda_handler
+from sqlalchemy import text
+
+
+OPT_IN_OUT = text("""SELECT va_profile_opt_in_out(:va_profile_id, :communication_item_id, :communication_channel_id, :allowed, :source_datetime);""")
+
+COUNT = r"""SELECT COUNT(*) FROM va_profile_local_cache;"""
+
+VA_PROFILE_TEST = text("""\
+SELECT allowed
+FROM va_profile_local_cache
+WHERE va_profile_id=:va_profile_id AND communication_item_id=:communication_item_id AND communication_channel_id=:communication_channel_id;""")
+
+
+def test_va_profile_cache_exists(notify_db):
+    assert notify_db.engine.has_table("va_profile_local_cache")
+
+
+def test_va_profile_opt_in_out(notify_db_session):
+    """
+    Test the stored function va_profile_opt_in_out by calling it directly.  The lambda function associated with
+    VA Profile integration calls this stored function.  The stored function should return True if any row was
+    created or updated; otherwise, False.
+    """
+
+    with notify_db_session.engine.begin() as connection:
+        # Begin with a sanity check.
+        result = connection.execute(COUNT)
+        assert result.fetchone()[0] == 0, "The cache should be empty at the start."
```

Comment:
Similar to the other comment about fetchone. Given the result is a select statement, this should absolutely be a rowcount.
```suggestion
        assert result.rowcount() == 0, "The cache should be empty at the start."
```